File: and-bible/src/net/bible/android/view/activity/readingplan/ReadingPlanItemAdapter.java
package net.bible.android.view.activity.readingplan;

import java.util.List;

import net.bible.service.readingplan.ReadingPlanInfoDto;
import android.content.Context;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ArrayAdapter;
import android.widget.TwoLineListItem;

/**
 * nice example here: http://shri.blog.kraya.co.uk/2010/04/19/android-multi-line-select-list/
 * 
 * @author Martin Denham [mjdenham at gmail dot com]
 * @see gnu.lgpl.License for license details.<br>
 *      The copyright to this program is held by it's author.
 */
public class ReadingPlanItemAdapter extends ArrayAdapter<ReadingPlanInfoDto> {

	private int resource;

	public ReadingPlanItemAdapter(Context _context, int _resource, List<ReadingPlanInfoDto> _items) {
		super(_context, _resource, _items);
		resource = _resource;
	}

	@Override
	public View getView(int position, View convertView, ViewGroup parent) {

		ReadingPlanInfoDto item = getItem(position);

		// Pick up the TwoLineListItem defined in the xml file
		TwoLineListItem view;
		if (convertView == null) {
			LayoutInflater inflater = (LayoutInflater) getContext().getSystemService(Context.LAYOUT_INFLATER_SERVICE);
			view = (TwoLineListItem) inflater.inflate(resource, parent, false);
		} else {
			view = (TwoLineListItem) convertView;
		}

		// Set value for the first text field
		if (view.getText1() != null) {
			String text = item.getCode();
			view.getText1().setText(text);
		}

		// set value for the second text field
		if (view.getText2() != null) {
			String text = item.getDescription();
			view.getText2().setText(text);
		}

		return view;
	}
}

File: app/src/main/java/net/bible/android/view/activity/readingplan/ReadingPlanItemAdapter.kt
/*
 * Copyright (c) 2020 Martin Denham, Tuomas Airaksinen and the And Bible contributors.
 *
 * This file is part of And Bible (http://github.com/AndBible/and-bible).
 *
 * And Bible is free software: you can redistribute it and/or modify it under the
 * terms of the GNU General Public License as published by the Free Software Foundation,
 * either version 3 of the License, or (at your option) any later version.
 *
 * And Bible is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
 * without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
 * See the GNU General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License along with And Bible.
 * If not, see http://www.gnu.org/licenses/.
 *
 */

package net.bible.android.view.activity.readingplan

import net.bible.service.readingplan.ReadingPlanInfoDto
import android.content.Context
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.ArrayAdapter
import android.widget.TwoLineListItem

/**
 * nice example here: http://shri.blog.kraya.co.uk/2010/04/19/android-multi-line-select-list/
 *
 * @author Martin Denham [mjdenham at gmail dot com]
 */
class ReadingPlanItemAdapter(_context: Context, private val resource: Int, _items: List<ReadingPlanInfoDto>):
		ArrayAdapter<ReadingPlanInfoDto>(_context, resource, _items) {

    override fun getView(position: Int, convertView: View?, parent: ViewGroup): View {

        val item = getItem(position)

        // Pick up the TwoLineListItem defined in the xml file
        val view: TwoLineListItem
        if (convertView == null) {
            val inflater = context.getSystemService(Context.LAYOUT_INFLATER_SERVICE) as LayoutInflater
            view = inflater.inflate(resource, parent, false) as TwoLineListItem
        } else {
            view = convertView as TwoLineListItem
        }

        // Set value for the first text field
        if (view.text1 != null) {
            val text = item!!.planName
            view.text1.text = text
        }

        // set value for the second text field
        if (view.text2 != null) {
            val text = item!!.planDescription
            view.text2.text = text
        }

        return view
    }
}
